libarchive/write.py

```diff
@@ -73,6 +73,47 @@ def add_files(self, *paths):
                         write_finish_entry(write_p)
                         entry_clear(entry_p)
 
+    def add_entry_from_memory(self, entry_path, entry_size, entry_data):
```

Comment:
Please rename this method to `add_file_from_memory`, inline the `_create_entry_archive` and `_close_entry_archive` methods into it, and add `filetype=REGULAR_FILE, permission=DEFAULT_UNIX_PERMISSION` to the arguments.